Code/Framework/AzToolsFramework/AzToolsFramework/Translation/Translation.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+ 
+#include <QApplication>
+#include <QTranslator>
+#include <QSettings>
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/Path/Path.h>
+ 
+namespace AzToolsFramework
+{
+    enum class Language
+    {
+        English,
+        Chinese
+    };
+ 
+    inline QTranslator* LoadTranslator(const QString& filename)
+    {
+        AZStd::array<char, AZ::IO::MaxPathLength> unresolvedPath;
+        QString dirpath = "@engroot@/Assets/Editor/Translations/" + filename;
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath(dirpath.toStdString().c_str(), unresolvedPath.data(), unresolvedPath.size());
```

Comment:
I would recommend using the AzCore/Utils/Utils.h [GetEngnePath](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L74-L77) function here.

That avoids needing the FileIOBase instance being initialized at this point.
Plus the code is a simpler to write.
```suggestion
        auto translationFilePath = (AZ::IO::FixedMaxPath(AZ::Utils::GetEnginePath()) / "Assets/Editor/Translation").LexicallyNormal();
```

Reply:
Afterwards a QString can be created using `QString::fromUtf8` from the AZ::IO::FixedMaxPath

```c++
auto translationFilePath = QString::fromUtf8(translationResolvedPath.c_str(), static_cast<int>(translationResolvedPath.Native().size()));
```